core/server/models/plugins/filter.js

```diff
@@ -191,36 +193,33 @@ filter = function filter(Bookshelf) {
         },
 
         /**
-         * ## Apply Filters
-         * Method which makes the necessary query builder calls (through knex) for the filters set
-         * on this model instance
-         * @param {Object} options
-         * @returns {Bookshelf.Model}
+         * @TODO:
+         *
+         * - release nql with latest mongo-knex
+         * - NQL must forward options to mongo-knex
          */
         applyDefaultAndCustomFilters: function applyDefaultAndCustomFilters(options) {
-            var self = this;
-
-            // @TODO figure out a better place/way to trigger loading filters
-            if (!this._filters) {
-                this.fetchAndCombineFilters(options);
```

Comment:
I think we might still need this or alternative method to combine all the filters together :thinking: ?

Reply:
yeah 100%.

no need to touch or refactor existing functions of this filter plugin?
requirement: we need filter combination. the rest depends on the utility.